File: homework1/shahid_maryam.dart
void main() {

  // 1. initialise
  var myArray =  {'age': 20, 'cat': 'Jem', 'kitten': 'Beans', 'fruit': 'Peach'};
  
  // 2. get value for a given key - cat
  print(myArray['cat']);  // prints Jem
  
  // 3. add a new element
  myArray['colour'] = 'Black';
  
  // 4. remove an element
  myArray.remove('age');
  
  // 5. modify value of an existing element
  myArray['colour'] = 'Yellow';
  
  // 6. search for the existence of a key
  print(myArray.containsKey('cat'));  // prints true
  print(myArray.containsKey('dog'));  // prints false
  
  // 7. search for the existence of a value 
  print(myArray.containsValue('Jem'));    // prints true
  print(myArray.containsValue('Orange')); // prints false
  
  // 8. print key-value pair through foo
  foo (key, value) {
    print('$key, $value');
  }
  
  myArray.forEach((key, value) => foo(key, value));
  
}

File: homework1/shahid_maryam.py
# 1. initialise 
myList = {'age': 20, 'cat':  'Jem', 'kitten': 'Beans', 'fruit': 'Peach'}

# 2. get value for a given key - cat
print(myList['cat'])

# 3. add a new element
myList['colour'] = 'Purple'

# 4. remove an element
del myList['age']

# 5. modify value of an existing element
myList['colour'] = 'Yellow'

# 6. search for the existence of a key
print('cat' in myList)
print('dog' in myList)

print('cat' in myList.keys())
print('dog' in myList.keys())

# 7. search for the existence of a value
print('Jem' in myList.values())
print('Maryam' in myList.values())

# 8. print key-value pair through foo
def foo(k, v):
  print(k, v)

for key, value in myList.items():
  foo(key,value)
